src/node/stage/approved.rs

```diff
@@ -180,24 +150,28 @@ impl Approved {
             .our_info()
             .elders
             .values()
-            .filter(|p2p_node| p2p_node.name() != core.name())
+            .filter(|p2p_node| p2p_node.name() != self.node_info.full_id.public_id().name())
             .map(P2pNode::peer_addr)
             .copied()
             .collect();
-        core.send_message_to_targets(&recipients, recipients.len(), message.to_bytes());
+        self.comm
+            .send_message_to_targets(&recipients, recipients.len(), message.to_bytes())
+            .await?;
+
+        // We need to relay it to ourself as well
+        // TODO: don't send it over the wire but just internally,
+        // making sure we don't create a recursive call.
+        self.comm
+            .send_message_to_target(&self.comm.our_connection_info()?, message.to_bytes())
+            .await?;
 
-        self.handle_unordered_vote(core, vote, proof_share)
+        Ok(())
```

Comment:
Just call `handle_unordered_vote` here just like it was before. No need to worry about recursion, because accumulated vote might trigger another vote, but it's not going to be the same vote.

Reply:
This does create a recursion, you can try it and the compiler will complain, async doesn't like recursion unless you Box the Future apparently, so we either need to use the macro @Yoga07 used in other place or remove the recursion. I think we should remove the recursion with the proposal about not accumulating messages we discussed, and we'll be fine.

Reply:
I see, didn't know there are issues with async recusrion. Good to know.